prevent 2 builds at the same time on the same box.

jfclere opened the pull request:
2 builds at the same time will crash both builds.

jfclere committed:
prevent 2 builds at the same time on the same box.

Karm commented on tasks/prep:
```diff
@@ -244,7 +244,21 @@ EOH
+        ps -p $MYPID
+        if [ $? -eq 0 ]
+        then
+          $ECHO "Another build is running if not remove $_xbrootdir"
```
```
-           $ECHO "Another build is running if not remove $_xbrootdir" 
+           $ECHO "Another build is running. Remove $_xbrootdir if it is not."
``` 

?

jfclere replied:
fixed.

jfclere committed:
fix typo.

jfclere commented:
Actually $PPID is not correct.
+++
xbprep: $(XBUILD)
        @echo "Attempting build of package $(NAME)-$(VERSION)-$(RELEASE) for $(HOST_OS)/$(BUILD_ARCH)"
        @bash $(XBUILD) --verbose $(BUILD_FLAGS) prep >build.log
+++
So PPID is the bash not the gmake :-(

jfclere committed:
find the gmake pid.

jfclere commented:
Should be good now.

jfclere committed:
prevent overwrite.
Note that has_force is in share/stddefs in case you want to suppress
the new logic.

jfclere committed:
Add arch i386...

gzaronikas reviewed (approved):
Changes in code and logic LGTM. Making a note here that with that changes one has to add --force in BUILD_FLAGS var upon calling make in order to bypass the tagbuild ovewrite check.
